src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -530,6 +530,18 @@ private static bool ByteArrayHasPrefix(byte[] byteArray, int dataLength, byte[]
             return true;
         }
 
+        private static byte[] GetPreamble(Encoding encoding)
+        {
+            foreach (EncodingPreamblePair pair in s_encodingPreamblePairs)
+            {
+                if (pair.Encoding.Equals(encoding))
```

Comment:
The implementation for these are pretty straightforward:
https://github.com/dotnet/coreclr/blob/4cf8a6b082d9bb1789facd996d8265d3908757b2/src/mscorlib/src/System/Text/UTF8Encoding.cs#L2394
https://github.com/dotnet/coreclr/blob/4cf8a6b082d9bb1789facd996d8265d3908757b2/src/mscorlib/src/System/Text/UTF32Encoding.cs#L1206
https://github.com/dotnet/coreclr/blob/4cf8a6b082d9bb1789facd996d8265d3908757b2/src/mscorlib/src/System/Text/UnicodeEncoding.cs#L1989

EncoderFallback/DecoderFallback implementations are similar.

Let me do some measurements.